pkg/odo/cli/pipelines/bootstrap.go

```diff
@@ -67,17 +70,32 @@ func (bo *BootstrapOptions) Validate() error {
 // Run runs the project bootstrap command.
 func (bo *BootstrapOptions) Run() error {
 	options := pipelines.BootstrapOptions{
-		DeploymentPath:   bo.deploymentPath,
-		GithubToken:      bo.githubToken,
-		GitRepo:          bo.gitRepo,
-		Prefix:           bo.prefix,
-		QuayAuthFileName: bo.quayIOAuthFilename,
-		QuayUserName:     bo.quayUsername,
-		SkipChecks:       bo.skipChecks,
+		DeploymentPath:           bo.deploymentPath,
+		GithubToken:              bo.githubToken,
+		GitRepo:                  bo.gitRepo,
+		ImageRepo:                bo.imageRepo,
+		InternalRegistryHostname: bo.internalRegistryHostname,
+		Prefix:                   bo.prefix,
+		QuayAuthFileName:         bo.quayIOAuthFilename,
+		QuayUserName:             bo.quayUsername,
+		SkipChecks:               bo.skipChecks,
+	}
+	if err := checkOptionalRegistryParams(options); err != nil {
+		return err
 	}
 	return pipelines.Bootstrap(&options)
 }
 
+func checkOptionalRegistryParams(options pipelines.BootstrapOptions) error {
```

Comment:
Shouldn't this only be checked if we're using Quay?

Reply:
Actually, quay user name is not needed anymore (now that we have --image-repo).    dockerconfigjson is checked after we parse image-repo and determine whether it is an internal or external registry.   We validate dockerconfigjson filename only if it is needed (using external registry).